sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs

```diff
@@ -207,40 +203,39 @@ private static void VerifyBuildErrors(ImmutableArray<Diagnostic> diagnostics, Co
         private static void VerifyNoExceptionThrown(IEnumerable<Diagnostic> diagnostics) =>
             diagnostics.Should().NotContain(d => d.Id == AnalyzerFailedDiagnosticId);
 
-        private static string VerifyPrimaryIssue(IList<IIssueLocation> expectedIssues, Func<IIssueLocation, bool> issueFilter,
-            Location location, string message, string extraInfo)
-        {
-            return VerifyIssue(expectedIssues, issueFilter, location, message, extraInfo, true, null);
-        }
+        private static string VerifyPrimaryIssue(Compilation compilation, IList<IIssueLocation> expectedIssues, Func<IIssueLocation, bool> issueFilter,
+            Location location, string message, string extraInfo) =>
+            VerifyIssue(compilation, expectedIssues, issueFilter, location, message, extraInfo, true, null);
 
-        private static void VerifySecondaryIssue(IList<IIssueLocation> expectedIssues, Func<IIssueLocation, bool> issueFilter,
-            Location location, string message, string issueId)
-        {
-            VerifyIssue(expectedIssues, issueFilter, location, message, null, false, issueId);
-        }
+        private static void VerifySecondaryIssue(Compilation compilation, IList<IIssueLocation> expectedIssues, Func<IIssueLocation, bool> issueFilter,
+            Location location, string message, string issueId) =>
+            VerifyIssue(compilation, expectedIssues, issueFilter, location, message, null, false, issueId);
 
-        private static string VerifyIssue(IList<IIssueLocation> expectedIssues, Func<IIssueLocation, bool> issueFilter,
+        private static string VerifyIssue(Compilation compilation, IList<IIssueLocation> expectedIssues, Func<IIssueLocation, bool> issueFilter,
```

Comment:
- This method needs only the language string and not the full compilation object so we should pass version instead.
- We can use the more generic type `ICollection<IIssueLocation>` instead of `IList<IIssueLocation>` for expectedIssues